web/src/lib/components/asset-viewer/asset-viewer.svelte

```diff
@@ -459,19 +506,56 @@
         on:toggleArchive={toggleArchive}
         on:asProfileImage={() => (isShowProfileImageCrop = true)}
         on:runJob={({ detail: job }) => handleRunJob(job)}
-        on:playSlideShow={handlePlaySlideshow}
+        on:playSlideShow={() => ($slideshowState = SlideshowState.PlaySlideshow)}
         on:unstack={handleUnstack}
       />
-    {/if}
-  </div>
+    </div>
+  {/if}
 
-  {#if !isSlideshowMode && showNavigation}
-    <div class="column-span-1 z-[999] col-start-1 row-span-1 row-start-2 mb-[60px] justify-self-start">
+  {#if $slideshowState === SlideshowState.None && showNavigation}
+    <div class="z-[1001] column-span-1 col-start-1 row-span-1 row-start-2 mb-[60px] justify-self-start">
       <NavigationArea on:click={navigateAssetBackward}><Icon path={mdiChevronLeft} size="36" /></NavigationArea>
     </div>
   {/if}
+
   <!-- Asset Viewer -->
-  <div class="col-span-4 col-start-1 row-span-full row-start-1">
+  <div class="z-[1000] col-start-1 col-span-4 row-start-1 row-span-full" bind:this={assetViewerHtmlElement}>
+    {#if $slideshowState != SlideshowState.None}
+      <!-- Slideshow actions bar -->
+      <div class="z-[1000] absolute w-full flex">
+        <div class="m-4 flex gap-2">
+          <CircleIconButton
+            icon={mdiClose}
+            on:click={() => ($slideshowState = SlideshowState.StopSlideshow)}
+            title="Exit Slideshow"
+          />
+          {#if $slideshowShuffle}
+            <CircleIconButton icon={mdiShuffle} on:click={() => ($slideshowShuffle = false)} title="Shuffle" />
+          {:else}
+            <CircleIconButton
+              icon={mdiShuffleDisabled}
+              on:click={() => ($slideshowShuffle = true)}
+              title="No shuffle"
+            />
+          {/if}
+          <CircleIconButton
+            icon={progressBarStatus === ProgressBarStatus.Paused ? mdiPlay : mdiPause}
+            on:click={() => (progressBarStatus === ProgressBarStatus.Paused ? progressBar.play() : progressBar.pause())}
+            title={progressBarStatus === ProgressBarStatus.Paused ? 'Play' : 'Pause'}
+          />
+          <CircleIconButton icon={mdiChevronLeft} on:click={navigateAssetBackward} title="Previous" />
+          <CircleIconButton icon={mdiChevronRight} on:click={navigateAssetForward} title="Next" />
+        </div>
+        <ProgressBar
+          autoplay
+          bind:this={progressBar}
+          bind:status={progressBarStatus}
+          on:done={navigateAssetForward}
+          duration={5000}
+        />
+      </div>
```

Comment:
I know the asset-viewer component is already _huge_. What do you think about moving all these elements into a separate slideshow-bar component (or similar)? Almost everything would work fine with just reading/writing to the slideshow store. The only thing that seems to be also needed is triggering the navigate asset forward and backwards. What do you think about emitting those events from the new component instead?

Reply:
Yes, it is indeed huge. In this PR, I've decided to group the viewer area and navigation controls (next/prev navigation buttons, slideshow bar) together. I have a plan to try to extract the "Viewer with basic navigation" component  in future PRs, leaving the top action navigation bar and details panel in the main component. 
I agree, it would be good to extract the "Slideshow bar" component in this PR. I'll try to do it later today